### PR TITLE
Set `deprecated` field on `CompletionItem`s

### PR DESCRIPTION
**Pull request by martskins**
This PR aims to address #2042 by setting the deprecated field for completion items.

The setting the tags field for LSP 3.15 part still needs fixing, but that one is blocked due to lsp-types not being up to date with 3.15 yet.

### CODE REVIEW

File: crates/ra_ide_api/src/completion/completion_item.rs [hunk 1]
```diff
@@ -44,6 +44,9 @@ pub struct CompletionItem {
     /// Additional info to show in the UI pop up.
     detail: Option<String>,
     documentation: Option<Documentation>,
+
+    /// Whether this item is marked as deprecated
+    deprecated: Option<bool>,
```
Comment (matklad) [inline crates/ra_ide_api/src/completion/completion_item.rs]: Why is this an option? Seems like `bool` is a better fit?

File: crates/ra_ide_api/src/completion/completion_item.rs [hunk 2]
```diff
@@ -70,6 +73,9 @@ impl fmt::Debug for CompletionItem {
         if let Some(documentation) = self.documentation() {
             s.field("documentation", &documentation);
         }
+        if let Some(deprecated) = self.deprecated {
+            s.field("deprecated", &deprecated);
```
Comment (matklad) [inline crates/ra_ide_api/src/completion/completion_item.rs]: let's not print anything unless the item is deprecated. That way, we won't be adding a new line to every completion item

File: crates/ra_ide_api/src/completion/completion_item.rs (hunk 1 above)
Comment (martskins) [reply crates/ra_ide_api/src/completion/completion_item.rs]: I agree. I was just trying to reflect the [specification](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_completion), where this field is listed as `deprecated?: boolean`, as closely as possible. But if that's not an issue, `bool` definitely sounds better.

File: crates/ra_ide_api/src/completion/completion_item.rs (hunk 2 above)
Comment (martskins) [reply crates/ra_ide_api/src/completion/completion_item.rs]: Makes sense, I'll do that.

File: crates/ra_ide_api/src/completion/completion_item.rs (hunk 1 above)
Comment (matklad) [reply crates/ra_ide_api/src/completion/completion_item.rs]: types inside ra_ide_api are explicitly *not* modeled after lsp. Rather, we use what makes most sense for the rust language, and explicitly lower this to lsp lowest common denominator in the conv.rs layer.

Comment (martskins) [reply crates/ra_ide_api/src/completion/completion_item.rs]: Ah ok, sorry about that. Should be fixed now!

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by matklad** [comment]
See also https://github.com/rust-analyzer/rust-analyzer/issues/2167

**Pull request merged**